backend/Events/NightTimeVoteEvents.js

```diff
@@ -0,0 +1,54 @@
+const RoleEnum = require('../domain/Enum/Role');
+const SuspectRevealDTO = require('../domain/DTO/response/SuspectRevealDTO');
+
+/**
+ * Event handlers and logic for `mafia-vote`, 'medic-vote' and `detective-vote`
+ * This involves the functionality for night-time voting, which includes:
+ * - Mafia choosing who to kill before the next round
+ * - Medics choosing who to save
+ * - Detectives choosing who to suspect
+ * @param {any} io
+ * @param {any} socket
+ * @param {MafiaGame} mafiaGame
+ */
+exports.loadNightTimeEvents = (io, socket, mafiaGame) => {
+    /**
+     * Handler for 'mafia-vote', sets the chosen player for the Mafia to the player specified in the message.
+     * In future iterations, this will be changed to tally votes from mafia members before setting the chosen player.
+     */
+    socket.on('mafia-vote', (mafiaVoteObj) => {
+        let room = socket.player.roomID;
+        room.voteHandler.addMafiaVote(
+            socket.player,
+            room.getPlayerByNickname(mafiaVoteObj.votingFor)
+        );
+    });
+
+    /**
+     * Handler for 'medic-vote', pretty much the same logic as the mafia vote, except it sets the chosen player for the Medics.
+     */
+    socket.on('medic-vote', (medicVoteObj) => {
+        let room = socket.player.roomID;
+        room.voteHandler.setMedicChosenPlayer(
+            room.getPlayerByNickname(mafiaVoteObj.votingFor)
+        );
+    });
+
+    /**
+     * Handler for detective vote. Retrieves the player specified in the message, checks their role, and replies with a
+     * SuspectRevealDTO that reveals whether the chosen player is Mafia or not.
+     */
+    socket.on('detective-vote', (detectiveVoteObj) => {
+        let room = socket.player.roomID;
+
+        let suspect = room.getPlayerByNickname(detectiveVoteObj.votingFor);
+
+        socket.emit(
+            'suspect-reveal',
+            new SuspectRevealDTO(
+                suspect.nickname,
+                suspect.role == RoleEnum.MAFIA
```

Comment:
Would it be possible to trigger a positive if the user was a Jester too? That would add a false positive and add a challenging aspect to the role of detective?

Reply:
Added this in a recent commit, "suspect-reveal" now has "isMafia" set to true if the suspect is a Jester